pkg/scale/decode.go

```diff
@@ -60,26 +60,19 @@ func indirect(dstv reflect.Value) (elem reflect.Value) {
 func Unmarshal(data []byte, dst interface{}) (err error) {
 	dstv := reflect.ValueOf(dst)
 	if dstv.Kind() != reflect.Ptr || dstv.IsNil() {
-		err = fmt.Errorf("unsupported dst: %T, must be a pointer to a destination", dst)
+		err = fmt.Errorf("%w: %T", ErrUnsupportedDestination, dst)
 		return
 	}
 
 	elem := indirect(dstv)
```

Comment:
In most cases, Variables should be created right before where they are needed. `elem` is used just once. I would not create the variable and directly use it.
So, in this case you could just do `	err = ds.unmarshal(indirect(dstv))`